Algorithms/Search/BinarySearcher.cs

```diff
@@ -0,0 +1,160 @@
+﻿using System;
+using System.Collections.Generic;
+
+
+namespace Algorithms.Search
+{
+    public static class BinarySearcher
```

Comment:
It would be a great idea if this searcher class implements the [`IEnumerator`](https://docs.microsoft.com/en-us/dotnet/api/system.collections.ienumerator?view=netframework-4.8) interface.

Reply:
I implemented IEnumerator\<T\> interface.